src/app/frontend/deploy/validprotocol_directive.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** The name of this directive. */
+export const validProtocolValidationKey = 'validProtocol';
+
+/**
+ * Validates that chosen protocol is valid for chosen service type.
+ *
+ * @param {!angular.$resource} $resource
+ * @param {!angular.$q} $q
+ * @return {!angular.Directive}
+ * @ngInject
+ */
+export default function validProtocolDirective($resource, $q) {
+  const isExternalParam = 'isExternal';
+
+  return {
+    restrict: 'A',
+    require: 'ngModel',
+    scope: {
+      [isExternalParam]: '=',
+    },
+    link: function(scope, element, attrs, ctrl) {
+      /** @type {!angular.NgModelController} */
+      let ngModelController = ctrl;
+      let externalChanged = false;
+
+      scope.$watch(isExternalParam, (newVal, oldVal) => {
+        externalChanged = newVal !== oldVal;
```

Comment:
Actually it is not always true. It is false on the page load. When te first time it's called.

![screenshot from 2016-01-20 14-55-17](https://cloud.githubusercontent.com/assets/2285385/12450784/0faf3d36-bf86-11e5-97b0-27e6cb106a60.png)

Without checks on page load validation is called 3 times. I wanted to avoid that. Now validation is called only when external service switch is clicked or protocol type is changed without additional validation when page is loaded.